server/schedule/region_scatterer.go

```diff
@@ -102,30 +112,56 @@ func (r *RegionScatterer) Scatter(region *core.RegionInfo) (*operator.Operator,
 }
 
 func (r *RegionScatterer) scatterRegion(region *core.RegionInfo) *operator.Operator {
-	stores := r.collectAvailableStores(region)
-	targetPeers := make(map[uint64]*metapb.Peer)
+	specialFilter := filter.NewSpecialEngineFilter(r.name)
+	var ordinaryPeers []*metapb.Peer
+	specialPeers := make(map[string][]*metapb.Peer)
+	// Group peers by the engine of their stores
 	for _, peer := range region.GetPeers() {
-		if len(stores) == 0 {
-			// Reset selected stores if we have no available stores.
-			r.selected.reset()
-			stores = r.collectAvailableStores(region)
+		store := r.cluster.GetStore(peer.GetStoreId())
+		if specialFilter.Target(r.cluster, store) {
```

Comment:
Should the targe of the `specialFilter` is special stores rather than ordinary store? or named to ordinaryFilter?

Reply:
OK. All names related to "filter" are confusing - -
Let me change it to follow the rule that the name of the filter is what it keeps.